definition_validator/validator.cc

```diff
@@ -597,10 +680,12 @@ class ValidateWalk {
             // Only validateAbstract for this class if we haven't already (we already have if this
             // is a `class << self` ClassDef)
             validateAbstract(ctx, sym);
+            validateRequiredAncestors(ctx, sym);
         }
         validateAbstract(ctx, singleton);
         validateFinal(ctx, sym, classDef);
         validateSealed(ctx, sym, classDef);
+        validateRequiredAncestors(ctx, singleton);
```

Comment:
We also check the singleton for when something `extend` a module with required ancestors.